### PR TITLE
Updated implementation statement for ac-05.md

### PR DESCRIPTION
**Pull request by rnafarrete**
### This pull request changes...

Updated implementation statement based on feedback from 18F and developers

### This pull request is ready to merge when...

- [ ] Meets acceptance criteria for the issue 
- [ ] Code is reviewed by someone other than the original author
- [ ] Code is tested 
- [ ] Experience is approved by UX  
- [ ] Meets [a11y checklist](https://github.com/HHS/TANF-app/blob/main/docs/a11y/how-18f-will-test-a11y.md)
- [ ] Meets [Raft's Manual QASP Checklist](https://teams.microsoft.com/l/file/B5285301-13CA-46FD-A723-AFFD578C224A?tenantId=d58addea-5053-4a80-8499-ba4d944910df&fileType=docx&objectUrl=https%3A%2F%2Fhhsgov.sharepoint.com%2Fsites%2FTANFDataPortalOFA%2FShared%20Documents%2FGeneral%2FManual%20Testing%20Practices_Issue173.docx&baseUrl=https%3A%2F%2Fhhsgov.sharepoint.com%2Fsites%2FTANFDataPortalOFA&serviceName=teams&threadId=19:f769bbcb029f4f02b55ae7fad90e310d@thread.skype&groupId=41f194a6-c1d3-4680-933e-c8ee7d17e287):lock: (only applicable for PR to main HHS)
- [ ] Documentation is updated 
  - OpenAPI  
  - Readme
  - Entity Relationship Diagram (ERD)

### CODE REVIEW

File: docs/controls/access_control/ac-05.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+* These users cannot provision new accounts and do not have access to the account management functionality.
+
+**Part b.**  
+
+**Developers**  
+  - Use GitHub for project planning as well as the hosting site for all code repositories.  
+  - Ensures all testing, linting and vulnerability scan checks are passed before committing code to the repository or deploy to any environment.  
+  - Ensures a consistent development environment by using Docker to set up a code based container 
+  - Develops code in Dev environment.  
+  - Currently, only a dev environment is in place. A staging environment (which could be called a test environment) and a production environment will be set up in future releases.  The process of dev, stage (testing) and release to production is planned for future releases.  However, all code and documentation committed to HHS has to be approved by the Product Manager and the HHS Tech Lead.
```
Comment (carltonsmith) [inline docs/controls/access_control/ac-05.md:30]: The ADR doc referenced above changes this also

File: docs/controls/access_control/ac-05.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+**Developers**  
+  - Use GitHub for project planning as well as the hosting site for all code repositories.  
+  - Ensures all testing, linting and vulnerability scan checks are passed before committing code to the repository or deploy to any environment.  
+  - Ensures a consistent development environment by using Docker to set up a code based container 
+  - Develops code in Dev environment.  
+  - Currently, only a dev environment is in place. A staging environment (which could be called a test environment) and a production environment will be set up in future releases.  The process of dev, stage (testing) and release to production is planned for future releases.  However, all code and documentation committed to HHS has to be approved by the Product Manager and the HHS Tech Lead.  
+
+**Users (OFA Admin and STTs)**
+  - approves new users, updates profile information, inactivates, and reactivates users
```
Comment (carltonsmith) [inline docs/controls/access_control/ac-05.md:33]: This is only OFA Admin users - Data Preppers (STTs) do not have access to this

File: docs/controls/access_control/ac-05.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  - reads data over assigned region
+  - able to upload and submit new TANF reports, replace and resubmit TANF reports  
+  - access (to be defined) allows the development of reports as required
```
Comment (carltonsmith) [inline docs/controls/access_control/ac-05.md:36]: Not sure what this means

File: docs/controls/access_control/ac-05.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+**Part c.**   
+Granting access to the TDRS system is performed by the OFA Admin through a request in the application.  Users request access to be able to create a profile and submit/view report in the application.
```
Comment (carltonsmith) [inline docs/controls/access_control/ac-05.md:39]: 1. TDRS should be TDP
2. Users create a profile as part of the process to request access. They don't need access to create a profile.

File: docs/controls/access_control/ac-05.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+Access to the TDRS system occurs in two different ways: through the Django admin interface and through the TDRS application interface.
```
Comment (carltonsmith) [inline docs/controls/access_control/ac-05.md:41]: TDRS should be TDP

File: docs/controls/access_control/ac-05.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  - access (to be defined) allows the development of reports as required
+
+**Part c.**   
+Granting access to the TDRS system is performed by the OFA Admin through a request in the application.  Users request access to be able to create a profile and submit/view report in the application.
+
+Access to the TDRS system occurs in two different ways: through the Django admin interface and through the TDRS application interface.  
+
+***Access to the Django admin interface***  
+
+Access directly to the Django interface occurs through a combination of flags and assigned permissions based on responsibilities.
```
Comment (carltonsmith) [inline docs/controls/access_control/ac-05.md:45]: We should specify that the `superuser` and `staff` flags are required

**Review by carltonsmith — CHANGES_REQUESTED** [review]
Thanks for delivering this @rnafarrete! A few revisions are needed, please see the comments.

File: docs/controls/access_control/ac-05.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+### TDP Implementation
+
+**Part a.**  
+Software development and maintenance is separated through the following:
+* Developers use GitHub for project planning as well as the hosting site for all code repositories.  Additionally, Circle CI is used for Continuous Integration / Continuous Delivery - it automates checks for making sure all automated testing, linting and vulnerability scanning checks pass before committing code to the repository or deploying to any environment.  Finally, Docker is used to set up a code based container environment to ensure the same environment is used during development as it is when deployed to the cloud.  Docker also automates the deploys themselves.  Currently, only a dev environment is in place. A staging environment (which could be called a test environment) and a production environment will be set up in future releases.
+* System owner manages the account access. Additionally, System owner provisions a new user account on request.
+
+Data creation, control, and security functions are completed through the TDP application by a user who has been authenticated and is active in the system. These roles include OFA admin and STT users.
```
[the remark this answers is not in the document]
Comment (rnafarrete) [reply docs/controls/access_control/ac-05.md]: Replaced STT Users with Data Preppers

File: docs/controls/access_control/ac-05.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+Data creation, control, and security functions are completed through the TDP application by a user who has been authenticated and is active in the system. These roles include OFA admin and STT users. 
+* STT/Data Preppers collect TANF data and transmit final data to OFA through the TANF application.
+* OFA admins have the ability to upload data on behalf of STTs and upload data files locally into the web application. 
+* These users cannot provision new accounts and do not have access to the account management functionality.
+
+**Part b.**  
+
+**Developers**  
+  - Use GitHub for project planning as well as the hosting site for all code repositories.  
+  - Ensures all testing, linting and vulnerability scan checks are passed before committing code to the repository or deploy to any environment.
```
[the remark this answers is not in the document]
Comment (rnafarrete) [reply docs/controls/access_control/ac-05.md]: replaced with "CircleCI ensures"

File: docs/controls/access_control/ac-05.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+* OFA admins have the ability to upload data on behalf of STTs and upload data files locally into the web application. 
+* These users cannot provision new accounts and do not have access to the account management functionality.
+
+**Part b.**  
+
+**Developers**  
+  - Use GitHub for project planning as well as the hosting site for all code repositories.  
+  - Ensures all testing, linting and vulnerability scan checks are passed before committing code to the repository or deploy to any environment.  
+  - Ensures a consistent development environment by using Docker to set up a code based container 
+  - Develops code in Dev environment.
```
[the remark this answers is not in the document]
Comment (rnafarrete) [reply docs/controls/access_control/ac-05.md]: Updated line 29 to this statement.

File: docs/controls/access_control/ac-05.md (hunk 2 above)
Comment (rnafarrete) [reply docs/controls/access_control/ac-05.md]: updated to state that only OFA Admin users have this ability.

File: docs/controls/access_control/ac-05.md (hunk 3 above)
Comment (rnafarrete) [reply docs/controls/access_control/ac-05.md]: Deleted...was a leftover from a draft statement

File: docs/controls/access_control/ac-05.md (hunk 4 above)
Comment (rnafarrete) [reply docs/controls/access_control/ac-05.md]: Updated to reflect the comment/feedback.

File: docs/controls/access_control/ac-05.md (hunk 5 above)
Comment (rnafarrete) [reply docs/controls/access_control/ac-05.md]: Changed to TDP

File: docs/controls/access_control/ac-05.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+* Developers use GitHub for project planning as well as the hosting site for all code repositories.  Additionally, Circle CI is used for Continuous Integration / Continuous Delivery - it automates checks for making sure all automated testing, linting and vulnerability scanning checks pass before committing code to the repository or deploying to any environment.  Finally, Docker is used to set up a code based container environment to ensure the same environment is used during development as it is when deployed to the cloud.  Docker also automates the deploys themselves.  Currently, only a dev environment is in place. A staging environment (which could be called a test environment) and a production environment will be set up in future releases.
+* System owner manages the account access. Additionally, System owner provisions a new user account on request.
+
+Data creation, control, and security functions are completed through the TDP application by a user who has been authenticated and is active in the system. These roles include OFA admin and STT users. 
+* STT/Data Preppers collect TANF data and transmit final data to OFA through the TANF application.
```
[the remark this answers is not in the document]
Comment (rnafarrete) [reply docs/controls/access_control/ac-05.md]: Updated based on comments.

File: docs/controls/access_control/ac-05.md (hunk 6 above)
Comment (rnafarrete) [reply docs/controls/access_control/ac-05.md]: Updated - added superuser and staff flags requirement

File: docs/controls/access_control/ac-05.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+b. Documents separation of duties of individuals; and 
+
+c. Defines information system access authorizations to support separation of duties.
+
+### TDP Implementation
+
+**Part a.**  
+Software development and maintenance is separated through the following:
+* Developers use GitHub for project planning as well as the hosting site for all code repositories.  Additionally, Circle CI is used for Continuous Integration / Continuous Delivery - it automates checks for making sure all automated testing, linting and vulnerability scanning checks pass before committing code to the repository or deploying to any environment.  Finally, Docker is used to set up a code based container environment to ensure the same environment is used during development as it is when deployed to the cloud.  Docker also automates the deploys themselves.  Currently, only a dev environment is in place. A staging environment (which could be called a test environment) and a production environment will be set up in future releases.
+* System owner manages the account access. Additionally, System owner provisions a new user account on request.
```
[the remark this answers is not in the document]
Comment (rnafarrete) [reply docs/controls/access_control/ac-05.md:16]: Updated to reflect the recent changes.

File: docs/controls/access_control/ac-05.md (hunk 1 above)
Comment (rnafarrete) [reply docs/controls/access_control/ac-05.md]: Updated to reflect change in environments

**Comment by rnafarrete** [comment]
Updated based on feedback/comments.

**Review by carltonsmith — APPROVED** [review]
LGTM!

**Comment by carltonsmith** [comment]
Pull request issued to HHS/main

**Pull request merged**